File: solutions/27/27.swift
class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        var k = 0;  // Number of elements not equal to val

        for i in 0..<nums.count {
            if nums[i] != val {
                nums[k] = nums[i]   // Move non-val element to the front
                k += 1  // Increment count of non-val elements
            }
        }

        return k
    }
}

File: solutions/27/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;  // Number of elements not equal to val

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[k] = nums[i];  // Move non-val element to the front
                k++;    // Increment count of non-val elements
            }
        }

        return k;    
    }
}